Sources/Turf/GeoJSON.swift

```diff
@@ -0,0 +1,358 @@
+import Foundation
+#if !os(Linux)
+import CoreLocation
+#endif
+
+
+public class BaseFeature: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case properties
+    }
+    
+    public var properties: [String : AnyJSONType]?
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        properties = try container.decode([String : AnyJSONType]?.self, forKey: .properties)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(properties, forKey: .properties)
+    }
+}
+
+public class Feature: BaseFeature {
+    public var type: GeoJSONType
+    
+    // Used to extract the geometry’s type w/o double decoding its coordinates
+    fileprivate var simplifiedGeometry: Geometry?
+    
+    private enum CodingKeys: String, CodingKey {
+        case type
+        //case properties
+        case simplifiedGeometry = "geometry"
+    }
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        simplifiedGeometry = try container.decodeIfPresent(Geometry.self, forKey: .simplifiedGeometry)
+        type = try GeoJSONType(rawValue: container.decode(String.self, forKey: .type))!
+        
+        try super.init(from: decoder)
+    }
+    
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(type.rawValue, forKey: .type)
+        try super.encode(to: encoder)
+    }
+}
+
+public struct Geometry: Codable {
+    public var type: String
+}
+
+/**
+ A `LineString` struct represents a shape consisting of two or more coordinates,
+ specified as `[CLLocationCoordinate2D]`
+ */
+public struct LineString: Codable {
```

Comment:
Maybe typealias Polyline to LineString (and MultiPolyline to MultiLineString), both for backwards compatibility and to accommodate muscle memory, especially since the map SDK still calls it MGLPolyline.

Reply:
Polyline to LineString compatibility alias should be sufficient since MultiPolyline is being introduced in this PR.

Reply:
OK, the map SDK has MGLMultiPolyline too, but I suppose that isn’t as well-known as MGLPolyline.

Reply:
For context, the main reason we originally named the class MGLPolyline instead of MGLLineString was for consistency with MapKit, but LineString also sounded too similar to String, especially in the context of an application that uses Core Text.